source/particle/property/elastic_stress.cc

```diff
@@ -0,0 +1,171 @@
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xx")));
+
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yy")));
+
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xy")));
+
+        if (dim == 3)
+          {
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_zz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yz")));
+          }
+
+      }
+
+      template <int dim>
+      void
+      ElasticStress<dim>::update_particle_property(const unsigned int data_position,
+                                                   const Vector<double> &solution,
+                                                   const std::vector<Tensor<1,dim> > &gradients,
+                                                   typename ParticleHandler<dim>::particle_iterator &particle) const
+      {
+        MaterialModel::MaterialModelInputs<dim> in(1, this->n_compositional_fields());
+
+        in.temperature[0] = solution[this->introspection().component_indices.temperature];
+
+        in.pressure[0] = solution[this->introspection().component_indices.pressure];
+
+        for (unsigned int d = 0; d < dim; ++d)
+          in.velocity[0][d] = solution[this->introspection().component_indices.velocities[d]];
+
+        for (unsigned int n = 0; n < this->n_compositional_fields(); ++n)
+          in.composition[0][n] = solution[this->introspection().component_indices.compositional_fields[n]];
+
+        Tensor<2,dim> grad_u;
+        for (unsigned int d=0; d<dim; ++d)
+          grad_u[d] = gradients[d];
+        in.strain_rate[0] = symmetrize (grad_u);
+
+        MaterialModel::MaterialModelOutputs<dim> out(1, this->n_compositional_fields());
+        this->get_material_model().evaluate (in,out);
+
+        for (unsigned int i = 0; i < SymmetricTensor<2,dim>::n_independent_components ; ++i)
+          {
+            particle->get_properties()[data_position + i] = solution[this->introspection().component_indices.compositional_fields[i]];
+          }
+      }
+
+      template <int dim>
+      UpdateTimeFlags
+      ElasticStress<dim>::need_update() const
+      {
+        return update_output_step;
+      }
+
+      template <int dim>
+      UpdateFlags
+      ElasticStress<dim>::get_needed_update_flags () const
+      {
+        return update_values;
```

Comment:
```suggestion
        return update_values | update_gradients;
```

Otherwise you can not ask for the strain rate in the update function.